js/coingecko.js

```diff
@@ -0,0 +1,219 @@
+'use strict';
+
+//  ---------------------------------------------------------------------------
+
+const Exchange = require ('./base/Exchange');
+
+//  ---------------------------------------------------------------------------
+
+module.exports = class coingecko extends Exchange {
+    describe () {
+        return this.deepExtend (super.describe (), {
+            'id': 'coingecko',
+            'name': 'CoinGecko',
+            'rateLimit': 67, // 15 requests per second
+            'version': 'v3',
+            'countries': [ 'SG' ],
+            'has': {
+                'CORS': false,
+                'publicAPI': true,
+                'privateAPI': false,
+                'cancelOrder': false,
+                'createDepositAddress': false,
+                'createOrder': false,
+                'deposit': false,
+                'fetchBalance': false,
+                'fetchClosedOrders': false,
+                'fetchCurrencies': true,
+                'fetchDepositAddress': false,
+                'fetchMarkets': true,
+                'fetchMyTrades': false,
+                'fetchOHLCV': false,
+                'fetchOpenOrders': false,
+                'fetchOrder': false,
+                'fetchOrderBook': false,
+                'fetchOrders': false,
+                'fetchTicker': true,
+                'fetchTickers': true,
+                'fetchBidsAsks': false,
+                'fetchTrades': false,
+                'withdraw': false,
+            },
+            'urls': {
+                'logo': 'https://user-images.githubusercontent.com/2256515/43684626-2318304a-98d6-11e8-9b80-e46e9d248976.png',
+                'api': {
+                    'public': 'https://api.coingecko.com/api',
+                },
+                'www': 'https://coingecko.com',
+                'doc': 'https://www.coingecko.com/api/docs/v3',
+            },
+            'requiredCredentials': {
+                'apiKey': false,
+                'secret': false,
+            },
+            'api': {
+                'public': {
+                    'get': [
+                        'coins/',
+                        'coins/list',
+                        'coins/markets',
+                    ],
+                },
+            },
+            'vsCurrencies': [
```

Comment:
Unfortunately, we can't instantiate this `vsCurrencies` variable, because it is not predefined in the base class. This will only work in JS, but not in Python nor in PHP. So, this array has to be moved to `this.options['vsCurrencies']`. The `options` property does exist everywhere already.

Reply:
Noted!